StorageAccountManagement/Program.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Azure;
+using Azure.Core;
+using Azure.Identity;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Storage;
+using Azure.ResourceManager.Storage.Models;
+
+using StorageAccountManagement;
+
+// Update placeholders with your own values - store values securely
+const string subscriptionId = "<subscription-id>";
+const string rgName = "<resource-group-name>";
+string storageAccountName = "<storage-account-name>";
+AzureLocation location = AzureLocation.EastUS;
+
+// Provide configuration options for the ArmClient
+ArmClientOptions armClientOptions = new()
+{
+    Retry = {
+        Delay = TimeSpan.FromSeconds(2),
+        MaxRetries = 5,
+        Mode = RetryMode.Exponential,
+        MaxDelay = TimeSpan.FromSeconds(10),
+        NetworkTimeout = TimeSpan.FromSeconds(100)
+    },
+};
+
+// Authenticate to Azure and create the top-level ArmClient
+ArmClient armClient = new(new DefaultAzureCredential(), subscriptionId, armClientOptions);
+
+// Create a resource identifier and get the subscription resource
+ResourceIdentifier resourceIdentifier = new($"/subscriptions/{subscriptionId}");
+SubscriptionResource subscription = armClient.GetSubscriptionResource(resourceIdentifier);
+
+// Register the Storage resource provider in the subscription
+await ManagementTasks.RegisterSRPInSubscription(subscription);
+
+// Create a new resource group - if one already exists then it's updated
+ArmOperation<ResourceGroupResource> rgOperation = await subscription
+    .GetResourceGroups()
+    .CreateOrUpdateAsync(WaitUntil.Completed, rgName, new ResourceGroupData(location));
+ResourceGroupResource resourceGroup = rgOperation.Value;
+
+// Check if the account name is available
+bool? nameAvailable = subscription
```

Comment:
We might should not continue to create storage account when the "nameAvailable" is false, or it will definitely fail.